Sources/Plasma/Apps/plPythonPack/main.cpp

```diff
@@ -169,21 +161,20 @@ void WritePythonFile(const plFileName &fileName, const plFileName &path, hsStrea
                 }
                 pythonCode = PythonInterface::CompileString(code, fileName);
                 hsAssert(pythonCode,"Not sure why this didn't compile the second time???");
-                fputs("an import file ", stdout);
+                ST::printf(out, "an import file ");
             }
             else
-                fputs("a PythonFile modifier(tm) ", stdout);
+                ST::printf(out, "a PythonFile modifier(tm) ");
         }
         else
         {
-            fputs("......blast! Error during run-code!\n", stdout);
+            ST::printf(stderr, "......blast! Error during run-code in {}!\n", fileName);
 
-            char* errmsg;
-            int chars_read = PythonInterface::getOutputAndReset(&errmsg);
-            if (chars_read > 0)
-            {
-                puts(errmsg);
-            }
+            ST::string msg;
+            PythonInterface::getOutputAndReset(msg);
+            ST::printf(out, "{}", msg);
+            PythonInterface::getErrorAndReset(msg);
+            ST::printf(stderr, "{}", msg);
```

Comment:
`puts()` (unlike `fputs()`) writes a terminating newline after its parameter.  Therefore, we still need to check for non-empty, and add a newline to the `printf` format...

Reply:
👍